[WIP] Spaces docs jim (#303)
OctopusDeploy/docs: docs/administration/spaces/index.md

## Managing Spaces

Spaces are managed by navigating to {{Configuration,Spaces}}.

An Octopus Administrator, or a team member with sufficient permission, is able to create, remove or modify spaces from this screen. It is also possible to [nominate a default space](#change-the-default-space), or [disable the default space entirely](#disable-the-default-space). Each space has a logo, which is also shown in the [space switcher](#switching-between-spaces) to make it easy to identify which space is currently focused upon by the UI. There is also a search filter to quickly find the spaces that you are interested in managing, this searches over the name and description fields of each space.

![Spaces configuration page](spaces-configuration.png)

### The Space Manager

Each space has a **Space Manager**. The space manager is the Administrator for that Space and is responsible for managing users and teams to the space and assigning permissions to them.

> When creating a new space, you are required to nominate a team member (or a team) to the role of space manager. This space manager is then responsible for [managing teams and permissions](/docs/managing-users-and-teams) within that space.

Reviewer: This paragraph can probably be combined with the previous one - mostly says the same thing.

> The user who creates a space doesn't necessarily need to be the space manager of the space. This enables a 'hands off' administrative approach suited to larger organizations or those who which to which to separate the duties of Octopus Server Administration, from the duties of Team Administration.

Reviewer: I don't like the word Team Administration here. Maybe something more like...

```suggestion
The user who creates a space doesn't necessarily need to be the space manager of the space. This enables a 'hands off' administrative approach suited to larger organizations where server administrators have no need to manage or deploy projects.
```

3. Give the Space a manager. This can be individual users or teams. Either can be selected from the drop-down menu. Click **SAVE**.
4. Provide a description for the Space.
5. Optionally, upload a logo for the Space.
6. Click **SAVE**.

![Add new space](add-new-space.png)

### Modify a Space

You can modify a space by navigating to {{Configuration,Spaces}} and selecting the space you want to modify. Expand the field you would like to change and click **SAVE** to save your changes.

Here you can rename a space, give it a new logo, modify the space managers, or even stop the spaces task queue from processing.

![Modify a space](modify-space.png)

### Delete a Space

You can delete spaces when you are the **Space manager**. Deleting a space cannot be undone, and the space and all of its contents, including projects, environments, releases, and deployment history will be deleted.

1. Navigate to {{Configuration,Spaces}} and select the space you want to delete.
1. Expand the **Task Queue Status** section and select the Stop task queue checkbox, and click **SAVE**.
1. Click the overflow button and select **Delete**.
1. Enter the name of the space and click **DELETE**.

> ## Change the Default Space {#change-the-default-space}

Reviewer: Probably worth mentioning the API in this section - say that the primary effect of the default space option is that any REST API calls where the space is not explicitly specified will be forwarded to the default space.

Reviewer: I know you've mentioned this below in the disable the default space section, but I think it should be up in this section.

Reviewer: Actually maybe a better way to structure this would be like:

```
## Default space

### Change or disable the default space
```

You can give some background about what the default space is and what it means for the API in the parent section. Changing or disabling the default space are cohesive because they have a similar effect on the API (potentially breaking existing scripts)

> The **Default space** is provided to existing installations as a mechanism to ensure that the instance operates in much the same way as it did prior to upgrading to a 'spaces enabled' build. Enabled by default, it's primary function is to provide an initial space for any existing artifacts, including our built in data at installation time, and effectively hides the existence of spaces until you are ready to start using them.

Reviewer: ```suggestion
The **Default space** is provided to existing installations as a mechanism to ensure that the instance operates in much the same way as it did prior to upgrading to a version of Octopus that supports Spaces. Enabled by default, it's primary function is to provide an initial space for any existing resources, and effectively hides the existence of spaces until you are ready to start using them.
```

It is possible to change the default space:

> Note: that the following carries some minor downtime for any automation that relies on the default space being available
 
1. Navigate to {{Configuration,Spaces}} and [disable the default space](#disable-the-default-space)
2. Return to {{Configuration,Spaces}} and select the space that you wish to nominate as the default space
3. Click the overflow button and select **Enable the default space**.

## Disable the Default Space {#disable-the-default-space}

For organizations that are new to Octopus, especially those that make heavy use of spaces, a default space is not necessarily required and therefore we have made it possible to remove the default space entirely. However, this comes with some considerations that should be weighed carefully against the needs of your organization.

> In addition to providing a home for existing resources, the default space allows any existing API calls that do not explicitly set a `Space Identifier` in the route, to be routed to the default space. For example, in the case where the default space ID is `Spaces-1` then the route `/api/projects/my-project` is equivalent to `/api/Spaces-1/projects/my-project`.

Reviewer: As I mentioned in another comment, I think this paragraph should be up in the `Change the Default Space` section

> With a default space enabled, any REST API calls that do not specify a space in the URL will be assumed to be directed to the default space. **By turning off the default space, this will no longer be the case. If you have a lot of bespoke automation relying on raw REST API calls, you will need to make changes to ensure these scripts explicitly specify the Space ID in the route. Otherwise they will break with the default space turned off.**

> This means that by disabling the default space - **you are opting into a non-backwards compatible scenario**, so be prepared! Things to check include:

Reviewer: Add Tentacle to this list

The following table shows what is space-scoped, system-scoped, or scoped to both.

| Resource              | Space-Scoped                | System-Scoped |
| --------------------- | --------------------------- | ------------- |
| Environments          | &#x2714;                    |               |
| Lifecycles            | &#x2714;                    |               |
| Projects              | &#x2714;                    |               |
| Variable Sets         | &#x2714;                    |               |
| Deployment Targets    | &#x2714;                    |               |
| Tenants               | &#x2714;                    |               |
| Octopus Server Nodes  |                             | &#x2714;      |
| Authentication        |                             | &#x2714;      |
| Users                 |                             | &#x2714;      |
| License               |                             | &#x2714;      |
| Events                | &#x2714;                    | &#x2714;      |

> | Teams                 | &#x2714;                    | &#x2714;      |

Reviewer: Add Tasks to this table too

Reviewer: We can add more things later. I think this list is fairly good for a starting point without us going to the effort of being exhaustive
